bin/unpublish-tags.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+dockerhub_token=$(curl -s -f -H "Content-Type: application/json" -X POST -d "{\"username\": \"${DOCKER_HUB_USERNAME}\", \"password\": \"${DOCKER_HUB_TOKEN}\"}" https://hub.docker.com/v2/users/login/ | jq -r .token)
+
+unpublish_group() {
+    local stackVersion="$1"
+    local targetTagSuffix="$2"
+    local status=0
+    variants=("" "-build")
+    if (( stackVersion <= 22 )); then
+        variants+=("-cnb" "-cnb-build")
+    fi
+    for variant in "${variants[@]}"; do
+      echo "Deleting heroku/heroku:${stackVersion}${variant}${targetTagSuffix}"
+        response=$(curl -s -X DELETE \
```

Comment:
> I saw an earlier version of this script used hub-tool - out of curiosity, what's the tradeoffs between that and manual curl calls?

I didn't love hub-tool, since it required another install, and is currently flagged as experimental. It also required a login separately from what we already have in `~/.docker/config.json`, which seemed to defeat the purpose of using it (why should I login to hub-tool when I'm already logged into docker via ~/.docker/config.json?). Most of the usage / docs I found seemed to favor actual humans logging in via the browser too. Also, `hub-tool` is somewhat challenging to search around for (I get a lot unrelated results for github actions).

> In the past I've used crane to remove tags

I should have know to check out crane. I've looked at it a few times over the course of this PR for manipulating manifests and annotations, but didn't end up using it. But at least for now, we don't have crane, and running curl against the api seems simple and fairly stable.

> I remember now - our API token was configured with reduced permissions that doesn't include delete (happy for us to change that though).

Yes, the token did not have `Delete`. I've generated a new token that includes `Delete` for this purpose.

Reply:
The token with `Delete` still doesn't do the trick. I'm getting `httperror 403: operation requires admin permissions`. Which I think means we'll need to elevate the bot's access on the repo. I'll go ahead and do that.

Reply:
Elevating the bot to `Admin` did the trick. Deleting temporary tags seem to be working now.